subsys/bluetooth/host/hci_core.c

```diff
@@ -7946,6 +7946,12 @@ static bool valid_adv_ext_param(const struct bt_le_adv_param *param)
 		}
 	}
 
+	if ((param->options & (BT_LE_ADV_OPT_DIR_MODE_LOW_DUTY |
+			       BT_LE_ADV_OPT_DIR_ADDR_RPA)) &&
```

Comment:
Why is it necessary to also check BT_LE_ADV_OPT_DIR_ADDR_RPA?

Reply:
When the application selects BT_LE_ADV_OPT_DIR_ADDR_RPA, the application intends to perform directed advertising. I would expect the zephyr host to return an error when the peer address is not specified.